### PR TITLE
RFC: various fixes for the latest MOI

### PR DESCRIPTION
**Pull request by odow**
- if two bound constraints exist, then the reduced cost of the variable is only non-zero on the bound that is binding.
- adds model name
- adds variable names
- adds constraint names
- sorts indexes by `.value` so that the list return is ordered by creation time
- adds functions for quadratic constraint primal/dual solutions
- documents the interface

Constraint names currently implemented via a `Dict{String, Any}` which is pretty bad. But leaving until it is demonstrated as a performance problem.

- [ ] Once this is merged we can move the repo to MOI.

### CODE REVIEW

**Comment by odow** [comment]
Reminder for me: add a SingleVariable objective (and test to MOI).

File: src/ref.jl [hunk 1]
```diff
@@ -18,70 +32,313 @@ lqs_supported_objectives(s) = []
...
+
+"""
+    lqs_delrows!(m, start_idx::Int, end_idx::Int)::Void
+
+Delete the SOS constraints `start_idx`, `start_idx+1`, ..., `end_idx` from
+the model `m`.
+"""
+function lqs_delsos!(m::LinQuadOptimizer, start_idx, end_idx) end
+
+# TODO(@joaquim): what is sertype. Why not a function?
```
Comment (odow) [inline src/ref.jl]: @joaquimg

File: src/ref.jl [hunk 2]
```diff
@@ -18,70 +32,313 @@ lqs_supported_objectives(s) = []
...
+
+"""
+    lqs_chgobjsen(m, sense::Symbol)::Void
+
+Change the optimization sense of the model `m` to `sense`. `sense` must be
+`:min` or `:max`.
+"""
+function lqs_chgobjsen!(m::LinQuadOptimizer, sense) end
+
+#TODO(@joaquimg): why is this not in-place?
```
Comment (odow) [inline src/ref.jl]: @joaquimg

File: src/ref.jl [hunk 3]
```diff
@@ -18,70 +32,313 @@ lqs_supported_objectives(s) = []
...
+function lqs_getqcpi!(m::LinQuadOptimizer, x) end
+
+"""
+    lqs_getobjval!(m)
+
+Get the objective value of the solved model `m`.
+"""
 function lqs_getobjval(m::LinQuadOptimizer) end
+
+# TODO(@joaquimg): what is this?
```
Comment (odow) [inline src/ref.jl]: @joaquimg

**Comment by odow** [comment]
The interface could probably do with a big clean up. 

Remove the `lqs_` prefix (#4), remove some of the redundant methods, and make the method names more explanatory.

**Comment by joaquimg** [comment]
@odow I was out, the PR ir great. YES lets work on the renaming

**Pull request merged**